src/GraphWebhooks-Core/Helpers/GraphServiceClientFactory.cs

```diff
@@ -0,0 +1,24 @@
+﻿/*
+ *  Copyright (c) Microsoft. All rights reserved. Licensed under the MIT license.
+ *  See LICENSE in the source repository root for complete license information.
+ */
+
+using Microsoft.Graph;
+using System.Net.Http.Headers;
+
+namespace GraphWebhooks_Core.Helpers
+{
+    public static class GraphServiceClientFactory
+    {
+        public static GraphServiceClient GetAuthenticatedGraphClient(string accessToken)
```

Comment:
Same remark.
have a delegate here: Action<string> might be better if you think that the graph service client would use the token for more than the token lifetime (typically 10 mins)

Reply:
Updated as below:

`
public static async Task<GraphServiceClient> GetAuthenticatedGraphClient(Func<Task<string>> acquireAccessToken)
        {
            // Fetch the access token
            string accessToken = await acquireAccessToken.Invoke();

            return new GraphServiceClient(new DelegateAuthenticationProvider(
                    async (requestMessage) =>
                    {
                        // Append the access token to the request.
                        requestMessage.Headers.Authorization = new AuthenticationHeaderValue(
                            Infrastructure.Constants.BearerAuthorizationScheme, accessToken);
                    }));
        }
`